solutions/cf-1418C.mdx

```diff
@@ -0,0 +1,83 @@
+---
+id: cf-1418C
+source: CF
+title: Mortal Kombat Tower
+author: Ryan Chou
+---
+[Official Editorial (C++)](https://codeforces.com/blog/entry/82673)
+
+## Alternate Solution - DP
+### Explanation
+
+If we define $\texttt{dp}[i][j]$ as the minimum number of skip points needed to get to the $i$th boss on turn $j$ 
+(our turn is zero), then our transitions would be:
+
+$\texttt{dp}[i + 1][0] =\min(\texttt{dp}[i + 1][0], \texttt{dp}[i][1] + \texttt{hard}[i])$
+
+$\texttt{dp}[i + 2][0] =\min(\texttt{dp}[i + 2][0], \texttt{dp}[i][1] + \texttt{hard}[i] + \texttt{hard}[i + 1])$
+
+$\texttt{dp}[i + 1][1] =\min(\texttt{dp}[i + 1][1], \texttt{dp}[i][0])$
+
+$\texttt{dp}[i + 2][1] =\min(\texttt{dp}[i + 2][1], \texttt{dp}[i][0])$
+
+This is because from each state, either the previous move, or the previous two moves could have been from the other player.
+
+So, the answer would be $\min(dp[N][0], dp[N][1])$.
+
+### Implementation
+
+**Time Complexity:** $\mathcal{O}(N)$
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+void solve() {
+	int n;
+	cin >> n;
+
+	vector<int> a(n);
+	for (int i = 0; i < n; i++) {
+		cin >> a[i];
+	}
+
+	/*
+	 * dp[i][j] = the min amt of skip points 
+	 * on turn i (your turn is 0), on the jth boss.
+	 */
+	vector<vector<int>> dp(2, vector<int>(n + 1, 1e9));
```

Comment:
use `INT32_MAX`

Reply:
wouldn't that overflow when doing the addition?

Reply:
hm
add a comment, `1e9 to prevent overflow`